v21.1/schema-design-table.md

```diff
@@ -0,0 +1,502 @@
+  <button class="filter-button" data-scope="cockroachcloud">CockroachCloud</button>
+</div>
+
+## Before you begin
+
+Before reading this page, do the following:
+
+<ul>
+  <li>
+    <a href="install-cockroachdb.html">Install CockroachDB.</a>
+  </li>
+  <li>
+    <a class="filter-content" data-scope="local" href="secure-a-cluster.html">Start a local CockroachDB cluster.</a>
+    <a class="filter-content" data-scope="cockroachcloud" href="cockroachcloud/create-your-cluster.html">Create a CockroachCloud cluster.</a>
+  </li>
+  <li>
+    <a href="schema-design-overview.html">Review the database schema objects.</a>
+  </li>
+  <li>
+    <a href="schema-design-database.html">Create a database.</a>
+  </li>
+  <li>
+    <a href="schema-design-schema.html">Create a user-defined schema.</a>
+  </li>
+</ul>
+
+## Create a table
+
+Tables are the [logical objects in a cluster](schema-design-overview.html#database-schema-objects) that store data sent from your application's persistence layer. Tables organize records of data in rows and columns.
+
+To create a table, use a [`CREATE TABLE` statement](create-table.html), following the best practices that we have listed in the following sections:
+
+- [Name a table](#name-a-table)
+- [Define columns](#define-columns)
+- [Select primary key columns](#select-primary-key-columns)
+- [Add additional constraints](#add-additional-constraints)
+- [Execute `CREATE TABLE` statements](#execute-create-table-statements)
+
+### Name a table
+
+Naming a table is the first step in table creation. `CREATE TABLE` statements generally take the form:
+
+~~~
+CREATE TABLE [schema_name].[table_name] (
+  [elements]
+  );
+~~~
+
+Where `schema_name` is the name of the user-defined schema, `table_name` is the name of the table, and `elements` is a comma-separated list of table elements, such as column definitions.
+
+#### Table naming best practices
+
+Here are some best practices to follow when naming tables:
+
+- Use a fully-qualified name (i.e., `CREATE TABLE schema_name.table_name`). If you do not specify the user-defined schema in the table name, CockroachDB will create the table in the preloaded `public` schema.
+- Use a table name that reflects the contents of the table. For example, for a table containing information about orders, you could use the name `orders` (as opposed to naming the table something like `table1`).
+
+#### Table naming example
+
+Suppose you want to create a table to store information about users of the [MovR](movr.html) platform.
+
+In a text editor, open the `dbinit.sql` file that you used to create the `movr` user-defined schema in [Create a User-defined Schema](schema-design-schema.html). Under the `CREATE SCHEMA` statement, add an empty `CREATE TABLE` statement for the `users` table.
+
+The file should now look something like this:
+
+{% include copy-clipboard.html %}
+~~~
+CREATE SCHEMA IF NOT EXISTS cockroachlabs;
+
+CREATE TABLE cockroachlabs.movr.users (
```

Comment:
I wasn't too sure about it, but it might make sense, since we are going to be adding indexes to the initialization file in the next page on adding secondary indexes, and then will likely re-run the same initialization file.

On that topic... the only issue with running the `dbinit.sql` file again as-is in the next page is that `CREATE TYPE` does not support `IF NOT EXISTS`. This reflects Postgres syntax (https://www.postgresql.org/docs/current/sql-createtype.html), so it probably isn't a priority to add that either.

In any case, I've added `IF NOT EXISTS` to all the `CREATE TABLE` examples on this page. But it's really only for the convenience of the example.

Reply:
Then maybe we could add a `DROP TYPE` at the beginning of `dbinit.sql`?

Reply:
That's a great idea. Will do so in a follow-up PR (or in the indexes chapter).